### PR TITLE
Tyr : Add ntfs2mimir in Tyr tasks

### PR DESCRIPTION
**Pull request by benoit-bst**

### CODE REVIEW

**Comment by benoit-bst** [comment]
https://jira.canaltp.fr/browse/NAVP-753

**Comment by benoit-bst** [comment]
Jenkins, test this please!

**Review by TeXitoi — APPROVED** [review]

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+
+
+def upgrade():
+    op.add_column('instance', sa.Column('import_ntfs_in_mimir', sa.Boolean(), nullable=True))
+    op.execute('update instance set import_ntfs_in_mimir=false')
+    op.alter_column('instance', 'import_ntfs_in_mimir', nullable=False)
+
+
+def downgrade():
+    op.drop_column('instance', 'import_ntfs_in_mimir')
```
Comment (xlqian) [inline source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: newline plz :)

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+down_revision = '3acdde6329db'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.add_column('instance', sa.Column('import_ntfs_in_mimir', sa.Boolean(), nullable=True))
+    op.execute('update instance set import_ntfs_in_mimir=false')
+    op.alter_column('instance', 'import_ntfs_in_mimir', nullable=False)
```
Comment (xlqian) [inline source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: can't you set `server_default`  to false directly?

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+# revision identifiers, used by Alembic.
+revision = '465a7431358a'
+down_revision = '3acdde6329db'
+
+from alembic import op
+import sqlalchemy as sa
+
+
+def upgrade():
+    op.add_column('instance', sa.Column('import_ntfs_in_mimir', sa.Boolean(), nullable=True))
```
Comment (xlqian) [inline source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: you can try this:

`op.add_column('instance', sa.Column('import_ntfs_in_mimir', sa.Boolean(), nullable=False, server_value='false'))`

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py [hunk 4]
```diff
@@ -0,0 +1,24 @@
+"""empty message
```
Comment (kinnou02) [inline source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: add a explanation

File: source/tyr/tyr/tasks.py [hunk 5]
```diff
@@ -120,16 +120,22 @@ def import_data(files, instance, backup_file, async=True, reload=True, custom_ou
...
         binarisation = [ed2nav.si(instance_config, job.id, custom_output_dir)]
         actions.append(chain(*binarisation))
-        if dataset.family_type == 'pt' and instance.import_stops_in_mimir:
-            # if we are loading pt data we might want to load the stops to autocomplete
-            actions.append(stops2mimir.si(instance_config, filename, job.id, dataset_uid=dataset.uid))
+        # Reload kraken with new data after binarisation (New .nav.lz4)
         if reload:
             actions.append(reload_data.si(instance_config, job.id))
+        # Import ntfs in Mimir
+        if dataset.family_type == 'pt' and instance.import_ntfs_in_mimir:
```
Comment (kinnou02) [inline source/tyr/tyr/tasks.py:133]: why did you move them?

File: source/tyr/tyr/binarisation.py [hunk 6]
```diff
@@ -602,7 +602,42 @@ def stops2mimir(self, instance_config, input, job_id=None, dataset_uid=None):
     try:
         res = launch_exec('stops2mimir', argv, logger)
         if res != 0:
-            raise ValueError('stops2mimir failed')
+            # Do not raise error because that it breaks celery tasks chain.
+            # stops2mimir have to be non-blocking
+            logger.error('stops2mimir failed')
```
Comment (kinnou02) [inline source/tyr/tyr/binarisation.py]: I like the idea of not failing if stop2mimir fail, but I don't like that noboby will see the failure.
It should not be done in this PR, but we will have to report failure like this some way.

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py [hunk 7]
```diff
@@ -0,0 +1,24 @@
...
+
+Revision ID: 465a7431358a
+Revises: 3acdde6329db
+Create Date: 2018-02-23 14:50:23.086805
+
+"""
+
+# revision identifiers, used by Alembic.
+revision = '465a7431358a'
+down_revision = '3acdde6329db'
```
Comment (kinnou02) [inline source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: I think you need to rebase, and check if your migration still work, new head is `242138b08c92`

**Review by kinnou02 — CHANGES_REQUESTED** [review]

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py (hunk 4 above)
Comment (benoit-bst) [reply source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: :+1: 
:white_check_mark:

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py (hunk 7 above)
Comment (benoit-bst) [reply source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: :+1: 
:white_check_mark:

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py (hunk 3 above)
Comment (benoit-bst) [reply source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: yes, :white_check_mark:

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py (hunk 2 above)
Comment (benoit-bst) [reply source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: Yes, directly is possible
:white_check_mark:

File: source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py (hunk 1 above)
Comment (benoit-bst) [reply source/tyr/migrations/versions/465a7431358a_add_import_ntfs_to_mimir.py]: :+1: 
:white_check_mark:

File: source/tyr/tyr/binarisation.py (hunk 6 above)
Comment (benoit-bst) [reply source/tyr/tyr/binarisation.py]: Okay, so I add a TODO for future PR.
:white_check_mark:

File: source/tyr/tyr/tasks.py (hunk 5 above)
Comment (benoit-bst) [reply source/tyr/tyr/tasks.py:133]: Just to reload after the binarization, not really useful, for convenience.

**Review by xlqian — APPROVED** [review]

**Comment by benoit-bst** [comment]
Jenkins, test this please!

**Comment by benoit-bst** [comment]
Jenkins, test this please!

**Review by kinnou02 — APPROVED** [review]
